component/src/test/java/org/wso2/extension/siddhi/io/rabbitmq/sink/RabbitMQSinkTestCase.java

```diff
@@ -210,96 +217,127 @@ public void rabbitmqHeaderPublishTest() throws Exception {
-
-     RabbitMQSinkTestUtil.consumer("tlstest", "fanout", false,
-             false, "", eventArrived, count);
-     executionPlanRuntime.start();
-
-     fooStream.send(new Object[]{"WSO2", 55.6f, 100L});
-     fooStream.send(new Object[]{"IBM", 75.6f, 100L});
-     fooStream.send(new Object[]{"WSO2", 57.6f, 100L});
-     Thread.sleep(10000);
 
-     count = RabbitMQSinkTestUtil.getCount();
-     eventArrived = RabbitMQSinkTestUtil.geteventArrived();
-     AssertJunit.assertEquals(3, count);
-     AssertJunit.assertTrue(eventArrived);
+    @Test(expectedExceptions = SiddhiAppValidationException.class)
+    public void rabbitmqWithoutExchangeNameSinkTest() {
+        log.info("---------------------------------------------------------------------------------------------");
+        log.info("RabbitMQ Sink test without exchange name");
+        log.info("---------------------------------------------------------------------------------------------");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        siddhiManager.createSiddhiAppRuntime(
+                "@App:name('TestExecutionPlan') " +
+                        "define stream FooStream1 (symbol string, price float, volume long); " +
+                        "@info(name = 'query1') " +
+                        "@sink(type ='rabbitmq', uri ='amqp://guest:guest@172.17.0.2:5672', " +
+                        "exchange.type='topic', " +
+                        "@map(type='xml'))" +
+                        "Define stream BarStream1 (symbol string, price float, volume long);" +
+                        "from FooStream1 select symbol, price, volume insert into BarStream1;");
+    }
 
-     executionPlanRuntime.shutdown();
+    @Test(expectedExceptions = SiddhiAppCreationException.class)
+    public void rabbitmqInvalidExchangeTypeSinkTest() {
+        log.info("---------------------------------------------------------------------------------------------");
+        log.info("RabbitMQ Sink test without exchange name");
+        log.info("---------------------------------------------------------------------------------------------");
+        SiddhiManager siddhiManager = new SiddhiManager();
+        siddhiManager.createSiddhiAppRuntime(
+                "@App:name('TestExecutionPlan') " +
+                        "define stream FooStream1 (symbol string, price float, volume long); " +
+                        "@info(name = 'query1') " +
+                        "@sink(type ='rabbitmq', uri ='amqp://guest:guest@172.17.0.2:5672', " +
+                        "exchange.name='testexchangetype', exchange.type='exchange', " +
+                        "@map(type='xml'))" +
+                        "Define stream BarStream1 (symbol string, price float, volume long);" +
+                        "from FooStream1 select symbol, price, volume insert into BarStream1;");
+    }
 
-     }*/
+    @Test
+    public void rabbitmqInvalidUriSinkTest() {
```

Comment:
If a log is getting printed at this point can you please assert the log atleast. Try following http://blog.diabol.se/?p=16. Else there in no point in this test rather than line coverage

Reply:
Apply for all other such testcases below

Reply:
fixed in 29ae151